sdk/servicebus/Azure.Messaging.ServiceBus/tests/Receiver/ReceiverLiveTests.cs

```diff
@@ -63,12 +67,15 @@ await using (var scope = await ServiceBusScope.CreateWithQueue(enablePartitionin
                 var receivedMessageCount = 0;
                 var messageEnum = messages.GetEnumerator();
 
-                foreach (var item in await receiver.ReceiveBatchAsync(messageCount))
+                while (receivedMessageCount < messageCount)
                 {
-                    receivedMessageCount++;
-                    messageEnum.MoveNext();
-                    Assert.AreEqual(messageEnum.Current.MessageId, item.MessageId);
-                    Assert.AreEqual(item.DeliveryCount, 1);
+                    foreach (var item in await receiver.ReceiveBatchAsync(messageCount))
+                    {
+                        receivedMessageCount++;
+                        messageEnum.MoveNext();
+                        Assert.AreEqual(messageEnum.Current.MessageId, item.MessageId);
+                        Assert.AreEqual(item.DeliveryCount, 1);
+                    }
```

Comment:
I don't think this will work as expected. 

Lets say 1st time we received 9 messages, it will set receivedMessageCount as 9. It will come again inside while loop and then foreach loop,  lets say this time we again received 9 messages then ` Assert.AreEqual(item.DeliveryCount, 1);` will fail because each time it will increase delivery count. And finally it will set receivedMessageCount as 18.  Now this time it will come outside of while loop but `Assert.AreEqual(messageCount, receivedMessageCount);` will also fail.